### PR TITLE
MCR-2991 Retrieve orcid mapping from configurable mycore classification

### PR DESCRIPTION
**Pull request by Possommi**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2991).

### CODE REVIEW

File: mycore-base/src/main/java/org/mycore/common/xml/MCRXMLFunctions.java [hunk 1]
```diff
@@ -860,6 +861,44 @@ public static String stripHtml(final String s) {
...
+            MCRLabel xmapping = label.get();
+            if (!xmapping.getText().contains(mappingPrefix)) {
+                return "";
+            }
+
+            Optional<String> result = Arrays.stream(xmapping.getText().split(" "))
+                .filter(mapping -> mapping.startsWith(mappingPrefix))
+                .findFirst();
+
+            return result.isPresent() ? result.get().split(":")[1] : null;
```
Comment (golsch) [inline mycore-base/src/main/java/org/mycore/common/xml/MCRXMLFunctions.java]: Null as return does not match the Javadoc or is inconsistent compared to the empty string as return.

File: mycore-orcid2/src/main/resources/components/orcid2/config/mycore.properties [hunk 2]
```diff
@@ -100,3 +100,9 @@ MCR.ORCID2.User.TrustedNameIdentifierTypes=orcid
 
 # Identifier types used to identify publications. If empty, all identifiers are used.
 MCR.ORCID2.Object.TrustedIdentifierTypes=
+
+# Set the classification where the mapping of the genre from and to ORCID can be retrieved, the mapping must be
+# provided in a label[@xml:lang='x-mapping'] with the following format orcid:<type at orcid>
+#MCR.ORCID2.Genre.Mapping.Classification                =
+MCR.ORCID2.Genre.Mapping.Classification.Mapping.Prefix  = orcid
```
Comment (golsch) [inline mycore-orcid2/src/main/resources/components/orcid2/config/mycore.properties]: "...Mapping...Mapping...", maybe just "MCR.ORCID2.Genre.Mapping.Classification.Prefix"

File: mycore-orcid2/src/main/resources/xsl/orcid2/v3/work2mcr.xsl [hunk 3]
```diff
@@ -1,87 +1,138 @@
...
+    label[@xml:lang='x-mapping']. The text attribute of that label contains the actual mapping like
+    $MCR.ORCID2.Genre.Mapping.Classification.Mapping.Prefix:$ORCID-Work-Type e.g. orcid:other
+
+    To determine the correct mods:genre of the host provide a mapping in a label[@xml:lang='x-hosts'] of each category.
+    The x-hosts attribute should only contain valid category ids with respect to the $MCR.ORCID2.Genre.Mapping.Classification.
+  -->
+  <xsl:param name="MCR.ORCID2.Genre.Mapping.Classification"/>
+  <xsl:param name="MCR.ORCID2.Genre.Mapping.Classification.Mapping.Prefix"/>
+  <xsl:param name="MCR.ORCID2.Genre.Mapping.Default.Genre"/>
+  <xsl:param name="MCR.ORCID2.Genre.Mapping.Standalone.xHost" select="'standalone'"/>
```
Comment (golsch) [inline mycore-orcid2/src/main/resources/xsl/orcid2/v3/work2mcr.xsl:24]: I would also document the property in the mycore.properties.

**Review by golsch — CHANGES_REQUESTED** [review]

File: mycore-orcid2/src/main/resources/xsl/orcid2/v3/work2mcr.xsl (hunk 3 above)
Comment (kkrebs) [reply mycore-orcid2/src/main/resources/xsl/orcid2/v3/work2mcr.xsl:24]: We already have a classification mapping mechanism in place.

**Comment by Possommi** [comment]
Will use ```org.mycore.mods.classification.MCRClassificationMappingEventHandler``` and implement custom ```work2mcr.xsl``` and ```mcr2work.xsl```

**Pull request closed without merging**